### ISSUE TITLE
Cannot find -lsgx_tstdcxx

### ISSUE THREAD

**Issue by BSDinis** [lsds/TaLoS#21]
I have Intel-SGX and the drivers installed on my linux machine 

However, I don't have libsgx_tstdcxx, but I have libsgx_tcxx.

If I make that change, though, the loader doesn't find any reference to __explicit_bzero_chk.

Is there anything I need to correct?

System Details
Thinkpad L460
```zsh
$ uname -a
Linux althea 5.2.8-200.fc30.x86_64 #1 SMP Sat Aug 10 13:21:39 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by BSDinis** [comment]
According to the Intel SGX Developer Reference, `libsgx_tstdcxx` is deprecated and should be replaced by `libsgx_tcxx`.

So the problem is the `__explicit_bzero_chk`. 
Not sure how to solve

**Comment by BSDinis** [comment]
https://download.01.org/intel-sgx/linux-2.6/docs/Intel_SGX_Developer_Reference_Linux_2.6_Open_Source.pdf

For reference, page

**Comment by plaublin** [comment]
Hi

We are aware of several problems related to newer versions of the SDK and gcc.

One idea (to be tried) is maybe to replace the call to `bzero()` by a call to `memset()`.
Please feel free to submit a pull request if you happen to solve the issue.

**Comment by BSDinis** [comment]
OK, another thing I have noticed is the use of -Wall and -Werror in production.

This is, in general, a bad practice, as it makes it harder for users to compile (especially with newer compiler versions); I'm going at it right now, I'll get back if find a solution

**Comment by plaublin** [comment]
TaLoS is a research prototype. This is not production code.

**Comment by BSDinis** [comment]
Fair enough, I mentioned production code as a proxy for "client" code, ie code to be compiled by users and not developers.

As an aside, do you have a roadmap for a production ready version (or integration in sgxssl) ?

**Comment by plaublin** [comment]
There is no roadmap for TaLoS. The authors of TaLoS either changed institution or are working on different projects.
Nevertheless I am trying to address the issues when I have time.

**Comment by plaublin** [comment]
Fixed by 64960e0 and a5b5d90
